data/console/test_node.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
+set -e
+
+test_result_wrapper(){
+  local VERSION=$1
+  local FILE=$2
+
+  local NODE="/usr/bin/node$VERSION"
+  local OUTPUT="/tmp/test_output"
+
+  # Run the test (using common and custom flags if present) and save the output
+  echo "Running $NODE $GLOBAL_FLAGS ${node_flags[$VERSION $FILE]}$FILE"
+  set +e
+  $NODE $GLOBAL_FLAGS ${node_flags[$VERSION $FILE]} $FILE &> $OUTPUT
+  RESULT=$?
+  set -e
+
+  # If test failed, keep track of failure and print out its output
+  if [ $RESULT -ne 0 ]; then
+    TEST_RESULT="failed"
+    echo "Node v$VERSION Test $FILE" >> $FAILED_TEST_LIST
+
+    echo "FAILED"
+    echo "Test Output:"
+    cat $OUTPUT
+  else
+    echo "OK"
+  fi
+  rm $OUTPUT
+}
+
+test_node_version(){
+  local VERSION="$1"
+
+  echo "Starting to test node version $VERSION"
+
+  # Get latest sources to make sure tests contain correct patches
+  zypper -n --no-gpg-checks si --repo node_sources -D "nodejs$VERSION"
+
+  local SOURCE_FILE=""
+  SOURCE_FILE=$(cd /usr/src/packages/SOURCES/ && find . -type f -iname "node-v$VERSION*.tar.xz")
+  SOURCE_FILE=${SOURCE_FILE:2} # strip ./ from name
+
+  if [ -z "$SOURCE_FILE" ]; then
+    echo "Can't find the sources of nodejs$VERSION"
+    exit 1
+  fi
+
+  local SOURCE_DIR
+  SOURCE_DIR=${SOURCE_FILE%.tar.xz} #remove extension from filename to get directory
+
+  # Unpack and apply patches to source
+  # Run the tests on the source using the installed binary
+  pushd "/usr/src/packages/SOURCES"
+    quilt setup "../SPECS/nodejs$VERSION.spec"
+    pushd "$SOURCE_DIR"
+      quilt push -a
+      # Run all test-crypto and test-tls available
+      for f in $(find test \( -path \*sequential\* -or -path \*parallel\* \) \( -name test-crypto-\* -or -name test-tls-\* \)); do
+        test_result_wrapper $VERSION $f
+      done
+    popd
+  popd
+
+  # Cleanup sources
+  rm -rf /usr/src/packages/SOURCES/*
+  rm -rf /usr/src/packages/SPECS/* 
+}
+
+main(){
+
+  local OS_VERSION="$1"
+
+  # Install dependencies to apply source patches and run tests
+  zypper -n in quilt rpm-build openssl-1_1
+
+  # Get list of each nodejs version found from default repo
+  local NODE_VERSIONS=$(zypper -n search nodejs | egrep -i 'nodejs[0-9]{1,2} ' | cut -d'|' -f 2 | tr -d ' '| tr -d 'nodejs' | sort -h | uniq)
+  local NODE_LATEST_VERSION=$(zypper -n search nodejs | egrep -i 'nodejs[0-9]{1,2} ' | cut -d'|' -f 2 | tr -d ' '| tr -d 'nodejs' | sort -h | uniq | tail -n1)
+
+  for v in $NODE_VERSIONS; do
+    echo "Found node version: $v"
+  done
+  echo "Will test only latest version: $NODE_LATEST_VERSION" 
+
+  # Install latest nodejs version
+  zypper -n in --no-recommends "nodejs$NODE_LATEST_VERSION"
+
+  # Add sources repo to have latest source patches
+  zypper -n --gpg-auto-import-keys ar -f "http://download.suse.de/ibs/home:/adamm:/node_test/$OS_VERSION/" node_sources
```

Comment:
test is relying on home repo which can be eventually deleted? no openSUSE support?

Reply:
I also would shy away from home repos, @AdamMajer could that be published somewhere else? (devel:languages:node?) so that openSUSE can be supported too

Reply:
It's just a link from the devel project (`devel:languages:nodejs`) on OBS that then builds against our :Update SLES repositories. There are no source changes.